omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,144 @@
+
+
+def get_params():
+    return EntryPointParameters(
+            tfs=dict(
+                required=True,
+                type=str,
+                help="TFS file containing the β* and error for "
+                     "B{1,2}{H,V}IP{1,5}"
+            ),
+            inplace=dict(
+                required=False,
+                action='store_true',
+                help="Add the luminosity imbalance in the source TFS file as a"
+                     "header"
+            )
+    )
+
+
+def _validate_tfs(df):
+    # First validate the labels, which correspond to the IP and Beam
+    expected_labels = set(LABELS)
+    if len(df[LABEL]) != 4 or set(df[LABEL]) != expected_labels:
+        for label in LABELS:
+            if len(df.loc[df[LABEL] == label]) > 1:
+                msg = f'Found label {label} several times. Expected only once.'
+                LOG.error(msg)
+                raise KeyError(msg)
+
+        msg = 'Expected labels in the TFS file not found. Expected labels:' \
+              f' {expected_labels}'
+        LOG.error(msg)
+        raise KeyError(msg)
+
+    # Validate the columns we need: BETSTAR{X,Y} and ERRBETSTAR{X,Y}
+    expected_columns = [f'{BETASTAR}{p}' for p in PLANES] + \
+                       [f'{ERR}{BETASTAR}{p}' for p in PLANES]
+    if not all([column in df.columns for column in expected_columns]):
+        msg = 'Expected columns in the TFS file not found. Expected ' \
+              f'columns: {expected_columns}'
+        LOG.error(msg)
+        raise KeyError(msg)
+
+
+def get_imbalance(df):
+    rel_error = {}
+    eff_beta = {}
+    for ip in IPS:
+        # Get the individual rows for the IP1 and IP5
+        row = df.loc[df[LABEL].str.startswith(ip)]
+
+        # Compute the average β* for each plane with both beams
+        # and the absolute errors for each plane
+        average_beta = {}
+        error = {}
+        for p in PLANES:
+            average_beta[p] = row.sum(axis=0)[f'{BETASTAR}{p}'] / 2
+            error[p] = 0.5 * sqrt((row[f'{ERR}{BETASTAR}{p}'] ** 2).sum(axis=0))
+
+        # Compute the relative error for this IP
+        rel_error[ip] = 0
+        for p in PLANES:
+            rel_error[ip] += (error[p] ** 2) / (average_beta[p] ** 2)
+        rel_error[ip] = 0.5 * sqrt(rel_error[ip])
+
+        # Get the effective beta
+        eff_beta[ip] = 1 / sqrt(1 / (reduce(mul, average_beta.values())))
+
+    # Compute the whole relative error
+    relative_error = sum(rel_error.values())
+
+    # And get the imbalance
+    imbalance = eff_beta[IPS[0]] / eff_beta[IPS[1]]
+
+    return {'imbalance': imbalance,
+            'relative_error': relative_error,
+            'eff_beta_ip1': eff_beta[IPS[0]],
+            'rel_error_ip1': rel_error[IPS[0]],
+            'eff_beta_ip5': eff_beta[IPS[1]],
+            'rel_error_ip5': rel_error[IPS[1]]
+            }
+
+
+def print_luminosity(res):
+    LOG.info('{:22s}: {}'.format("Luminosity imbalance", res['imbalance']))
```

Comment:
what happened to f-strings? ^^'

Reply:
they were never here in the first place, they're in Urlaub